vkcs/compute/resource_volume_attach_test.go

```diff
@@ -25,6 +25,47 @@ func TestAccComputeVolumeAttach_basic(t *testing.T) {
 				Config: acctest.AccTestRenderConfig(testAccComputeVolumeAttachBasic),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckComputeVolumeAttachExists("vkcs_compute_volume_attach.va_1", &va),
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "metadata.key1", "val0"),
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "metadata.key2", "val2"),
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "metadata.key3", "val3"),
+
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "all_metadata.key1", "val0"),
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "all_metadata.key2", "val2"),
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "all_metadata.key3", "val3"),
+				),
+			},
+			{
+				Config: acctest.AccTestRenderConfig(testAccComputeVolumeAttachBasicMetadataUpdate),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeVolumeAttachExists("vkcs_compute_volume_attach.va_1", &va),
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "metadata.key1", "val1"),
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "metadata.key3", "val3"),
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "metadata.key4", "val4"),
+					resource.TestCheckNoResourceAttr("vkcs_blockstorage_volume.volume_1", "metadata.key2"),
+
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "all_metadata.key1", "val1"),
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "all_metadata.key3", "val3"),
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "all_metadata.key4", "val4"),
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "all_metadata.attached_mode", "rw"),
+					resource.TestCheckResourceAttr("vkcs_blockstorage_volume.volume_1", "all_metadata.readonly", "False"),
+					resource.TestCheckNoResourceAttr("vkcs_blockstorage_volume.volume_1", "all_metadata.key2"),
+				),
+			},
+			{
+				Config: acctest.AccTestRenderConfig(testAccComputeVolumeAttachBasicMetadataDelete),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeVolumeAttachExists("vkcs_compute_volume_attach.va_1", &va),
+					resource.TestCheckNoResourceAttr("vkcs_blockstorage_volume.volume_1", "metadata.key1"),
```

Comment:
Same with `testAccCheckBlockStorageVolumeEmptyMetadata`.

Reply:
use same function for all_metadata field